CMakeLists.txt

```diff
@@ -492,6 +494,15 @@ if (ONLY_SHARED_LIBS)
   set (H5_ENABLE_STATIC_LIB NO)
   set (BUILD_SHARED_LIBS ON CACHE BOOL "Build Shared Libraries" FORCE)
   set (BUILD_STATIC_LIBS OFF CACHE BOOL "Build Static Libraries" FORCE)
+  if (HDF5_BUILD_STATIC_TOOLS)
+    message (WARNING "Cannot build static tools without static libraries. Building shared tools.")
+  endif ()
+  set (HDF5_BUILD_STATIC_TOOLS OFF CACHE BOOL "Build Static Tools NOT Shared Tools" FORCE)
+endif ()
+
+if (NOT BUILD_SHARED_LIBS AND NOT HDF5_BUILD_STATIC_TOOLS)
```

Comment:
Since the default is `OFF` for `HDF5_BUILD_STATIC_TOOLS`, it seems like anyone who builds only a static library by turning off `BUILD_SHARED_LIBS` is going to get this warning by default, which could be either annoying or surprising/confusing. Since the only case I can think of where someone will want to set the `HDF5_BUILD_STATIC_TOOLS` option to on is the case where both static and shared are built but they want static tools instead, I think we probably don't need this warning.